percona/controllers/replica/replica.go

```diff
@@ -0,0 +1,320 @@
+	if reflect.DeepEqual(oldCluster.Spec.PGReplicas, newCluster.Spec.PGReplicas) {
+		return nil
+	}
+
+	oldReplicaCount := 0
+	newReplicaCount := 0
+	var err error
+
+	if newCluster.Spec.PGReplicas != nil {
+		newReplicaCount = newCluster.Spec.PGReplicas.HotStandby.Size
+	}
+	if oldCluster.Spec.PGReplicas != nil {
+		oldReplicaCount = oldCluster.Spec.PGReplicas.HotStandby.Size
+	}
+
+	if newReplicaCount == 0 {
+		for i := 1; i <= oldReplicaCount; i++ {
+			replicaName := getReplicaName(oldCluster, i)
+			err = clientset.CrunchydataV1().Pgreplicas(newCluster.Namespace).Delete(ctx, replicaName, metav1.DeleteOptions{})
+			if err != nil {
+				return errors.Wrapf(err, "delete replica %s", replicaName)
+			}
+		}
+		return nil
+	}
+	if newReplicaCount < oldReplicaCount {
+		for i := oldReplicaCount; i > newReplicaCount; i-- {
+			replicaName := getReplicaName(oldCluster, i)
+			err = clientset.CrunchydataV1().Pgreplicas(newCluster.Namespace).Delete(ctx, replicaName, metav1.DeleteOptions{})
+			if err != nil {
+				return errors.Wrapf(err, "delete replica %s", replicaName)
+			}
+		}
+	}
+
+	for i := 1; i <= newReplicaCount; i++ {
+		replicaName := getReplicaName(oldCluster, i)
+		oldReplica, err := clientset.CrunchydataV1().Pgreplicas(newCluster.Namespace).Get(ctx, replicaName, metav1.GetOptions{})
+		if err != nil {
+			replica := getNewReplicaObject(newCluster, &crv1.Pgreplica{}, i)
+			_, err = clientset.CrunchydataV1().Pgreplicas(newCluster.Namespace).Create(ctx, replica, metav1.CreateOptions{})
+			if err != nil {
+				return errors.Wrapf(err, "create replica %s", replica.Name)
+			}
+			continue
+		}
+		replica := getNewReplicaObject(newCluster, oldReplica, i)
+		if reflect.DeepEqual(newCluster.Spec.PGReplicas.HotStandby, oldCluster.Spec.PGReplicas.HotStandby) {
+			continue
+		}
+
+		replica.ResourceVersion = oldReplica.ResourceVersion
+		replica.Status = oldReplica.Status
+		_, err = clientset.CrunchydataV1().Pgreplicas(newCluster.Namespace).Update(ctx, replica, metav1.UpdateOptions{})
+		if err != nil {
+			return errors.Wrapf(err, "update replica %s", replica.Name)
+		}
+
+	}
+	log.Println("Handle service update")
+	err = createOrUpdateReplicaService(clientset, newCluster)
+	if err != nil {
+		return errors.Wrap(err, "handle replica service")
+	}
+	return nil
+}
+
+func getReplicaName(cluster *crv1.PerconaPGCluster, index int) string {
+	return cluster.Name + "-repl" + strconv.Itoa(index)
+}
+
+func getNewReplicaObject(cluster *crv1.PerconaPGCluster, replica *crv1.Pgreplica, index int) *crv1.Pgreplica {
+	if cluster.Spec.PGReplicas == nil {
+		return nil
+	}
+	labels := make(map[string]string)
+	labels[config.LABEL_PG_CLUSTER] = cluster.Name
+	labels[config.LABEL_NAME] = cluster.Name + "-repl" + strconv.Itoa(index)
+	labels["pgouser"] = "admin"
+	if cluster.Spec.PGReplicas.HotStandby.Labels != nil {
+		for k, v := range cluster.Spec.PGReplicas.HotStandby.Labels {
+			labels[k] = v
+		}
+	}
+	annotations := make(map[string]string)
```

Comment:
since we are not using them anywhere else except line 155, we can directly iterate over annotations.
We don't need that lines at all